File: src/services/tauri/platform.ts
import { invoke } from "@tauri-apps/api/core";

/** Antigravity installation status returned from the backend. */
export interface AntigravityInfo {
  isInstalled: boolean;
  configPath: string | null;
  isRunning: boolean;
  version: string | null;
}

/**
 * Tauri IPC wrapper for platform detection commands.
 */

/** Detects whether Antigravity is installed and running. */
export async function detectAntigravity(): Promise<AntigravityInfo> {
  return invoke<AntigravityInfo>("detect_antigravity");
}

/** Checks if Antigravity is currently running. */
export async function getAntigravityStatus(): Promise<boolean> {
  return invoke<boolean>("get_antigravity_status");
}

File: src-tauri/src/commands/platform.rs
//! Antigravity platform detection IPC commands.
//!
//! Provides information about the local Antigravity installation
//! status to the frontend.

use crate::error::AppError;
use crate::services::antigravity;

/// Detects whether Antigravity is installed and running.
///
/// Called from frontend: `invoke("detect_antigravity")`
#[tauri::command]
pub fn detect_antigravity() -> Result<antigravity::AntigravityInfo, AppError> {
    Ok(antigravity::detect_installation())
}

/// Checks if Antigravity is currently running.
///
/// Called from frontend: `invoke("get_antigravity_status")`
#[tauri::command]
pub fn get_antigravity_status() -> Result<bool, AppError> {
    Ok(antigravity::is_antigravity_running())
}
